fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/MergeAggregate.java

```diff
@@ -0,0 +1,195 @@
+        return (NamedExpression) e.rewriteDownShortCircuit(expr -> {
+            if (expr instanceof Alias && ((Alias) expr).child() instanceof AggregateFunction) {
+                Alias alias = (Alias) expr;
+                AggregateFunction aggFunc = (AggregateFunction) alias.child();
+                ExprId childExprId = ((SlotReference) aggFunc.child(0)).getExprId();
+                if (innerAggExprIdToAggFunc.containsKey(childExprId)) {
+                    return new Alias(alias.getExprId(), innerAggExprIdToAggFunc.get(childExprId),
+                            alias.getName());
+                } else {
+                    return expr;
+                }
+            } else {
+                return expr;
+            }
+        });
+    }
+
+    private boolean canMergeAggregate(Plan plan, boolean hasProject) {
+        LogicalAggregate<Plan> outerAgg = (LogicalAggregate<Plan>) plan;
+        LogicalAggregate<Plan> innerAgg = hasProject ? (LogicalAggregate<Plan>) outerAgg.child().child(0)
+                : (LogicalAggregate<Plan>) outerAgg.child();
+
+        List<Expression> outerAggGroupByKeys = outerAgg.getGroupByExpressions();
+        if (hasProject) {
+            LogicalProject<Plan> project = (LogicalProject<Plan>) outerAgg.child();
+            outerAggGroupByKeys = PlanUtils.replaceExpressionByProjections(project.getProjects(), outerAggGroupByKeys);
+        }
+
+        if (!new HashSet<>(innerAgg.getGroupByExpressions()).containsAll(outerAggGroupByKeys)) {
+            return false;
+        }
+
+        Map<ExprId, AggregateFunction> innerAggExprIdToAggFunc = innerAgg.getOutputExpressions().stream()
+                .filter(expr -> (expr instanceof Alias) && (expr.child(0) instanceof AggregateFunction))
+                .collect(Collectors.toMap(NamedExpression::getExprId, value -> (AggregateFunction) value.child(0)));
+        Set<AggregateFunction> aggregateFunctions = outerAgg.getAggregateFunctions();
+        for (AggregateFunction outerFunc : aggregateFunctions) {
+            if (!(ALLOW_MERGE_AGGREGATE_FUNCTIONS.contains(outerFunc.getName()))) {
+                return false;
+            }
+            if (outerFunc.isDistinct()) {
+                return false;
+            }
+            // not support outerAggFunc: sum(a+1),sum(a+b)
+            if (!(outerFunc.child(0) instanceof SlotReference)) {
+                return false;
+            }
+            ExprId childExprId = ((SlotReference) outerFunc.child(0)).getExprId();
+            if (innerAggExprIdToAggFunc.containsKey(childExprId)) {
+                AggregateFunction innerFunc = innerAggExprIdToAggFunc.get(childExprId);
+                if (innerFunc.isDistinct()) {
+                    return false;
+                }
+                // support sum(sum),min(min),max(max),sum(count),any_value(any_value)
+                if (!(outerFunc.getName().equals("sum") && innerFunc.getName().equals("count"))
```

Comment:
trans `sum(count())` to `count()` lead to nullable changed if outer agg is scalar agg. so we need to wrap the final expression with `nullable()` function to change its nullable to true